### PR TITLE
Java v4.32.0 release notes

### PR DESCRIPTION
**Pull request by newtork**
## What Has Changed?

Added release notes for Cloud SDK for Java 4.32.0

### CODE REVIEW

File: docs-java_versioned_docs/version-v4/release-notes/release-notes-15-to-29.mdx [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+      - Update [Apache HttpClient v5](https://central.sonatype.com/artifact/org.apache.httpcomponents.client5/httpclient5/5.3.1) (`org.apache.httpcomponents.client5:httpclient5`) from `5.2.2` to `5.3.1`
+      - Update [Apache HttpCore v5](https://central.sonatype.com/artifact/org.apache.httpcomponents.core5/httpcore5/5.2.4) (`org.apache.httpcomponents.core5:httpcore5`) from `5.2.3` to `5.2.4`
+      - Update [Commons IO](https://central.sonatype.com/artifact/commons-io/commons-io/2.15.1) (`commons-io:commons-io`) from `2.15.0` to `2.15.1`
+      - Update [Guava](https://central.sonatype.com/artifact/com.google.guava/guava/33.2.0-jre) (`com.google.guava:guava`) from `32.1.3-jre` to `33.2.0-jre`
+      - Update [Spring Framework](https://central.sonatype.com/artifact/org.springframework/spring-framework-bom/5.3.34) (`org.springframework:spring-framework-bom`) from `5.3.32` to `5.3.34`
+
+### Fixes issues
+
+- Fix an issue where the current principal wasn't correctly derived from the incoming request in BTP Neo applications.
```
Comment (newtork) [inline docs-java_versioned_docs/version-v4/release-notes/release-notes-15-to-29.mdx:31]:
```suggestion
- Fix an issue where the current principal wasn't correctly derived from the incoming request in SAP BTP Neo applications.
```

File: docs-java_versioned_docs/version-v4/release-notes.mdx [hunk 2]
```diff
@@ -32,13 +32,13 @@ It will help you:
...
 
-:::info Update to the SAP Cloud SDK v4
+:::info Update to the SAP Cloud SDK v5
 
-Version 3.x of the SAP Cloud SDK for Java has reached its end of life.
-As a consequence, there won't be any updates to the SAP Cloud SDK version 3 - not even security fixes.
-To continue using the latest features outlined in the release notes below, please [update to **the SAP Cloud SDK v4**](/docs/java/v4/guides/4.0-upgrade).
+Version 3.x and 4.x of the SAP Cloud SDK for Java has reached its end of life.
+As a consequence, there won't be any updates to the SAP Cloud SDK version 3 or 4 - not even security fixes.
+To continue using the latest features outlined in the release notes below, please [update to **the SAP Cloud SDK v4**](/docs/java/v4/guides/4.0-upgrade) and [to **the SAP Cloud SDK v5**](https://sap.github.io/cloud-sdk/docs/java/guides/5.0-upgrade-steps).
```
Comment (CharlesDuboisSAP) [inline docs-java_versioned_docs/version-v4/release-notes.mdx:41]: Suggestion:

### The SAP Cloud SDK v3 and v4 for Java are Deprecated

:::info Update to the SAP Cloud SDK v5

Version 3.x and 4.x of the SAP Cloud SDK for Java have reached their end of life.
As a consequence, there won't be any updates to the SAP Cloud SDK version 3 or 4 - not even security fixes.
To continue using the latest features outlined in the release notes below, please [update to **the SAP Cloud SDK v4**](/docs/java/v4/guides/4.0-upgrade) then [to **the SAP Cloud SDK v5**](https://sap.github.io/cloud-sdk/docs/java/guides/5.0-upgrade-steps).

File: docs-java_versioned_docs/version-v4/release-notes/release-notes-15-to-29.mdx [hunk 3]
```diff
@@ -1,3 +1,34 @@
...
+  - Other dependency updates:
+    - Minor version updates:
+      - Update [Jackson](https://central.sonatype.com/artifact/com.fasterxml.jackson.core/jackson-databind/2.17.0) (`com.fasterxml.jackson.core:jackson-databind`) from `2.16.0` to `2.17.0`
+      - Update [Apache HttpClient v5](https://central.sonatype.com/artifact/org.apache.httpcomponents.client5/httpclient5/5.3.1) (`org.apache.httpcomponents.client5:httpclient5`) from `5.2.2` to `5.3.1`
+      - Update [Apache HttpCore v5](https://central.sonatype.com/artifact/org.apache.httpcomponents.core5/httpcore5/5.2.4) (`org.apache.httpcomponents.core5:httpcore5`) from `5.2.3` to `5.2.4`
+      - Update [Commons IO](https://central.sonatype.com/artifact/commons-io/commons-io/2.15.1) (`commons-io:commons-io`) from `2.15.0` to `2.15.1`
+      - Update [Guava](https://central.sonatype.com/artifact/com.google.guava/guava/33.2.0-jre) (`com.google.guava:guava`) from `32.1.3-jre` to `33.2.0-jre`
+      - Update [Spring Framework](https://central.sonatype.com/artifact/org.springframework/spring-framework-bom/5.3.34) (`org.springframework:spring-framework-bom`) from `5.3.32` to `5.3.34`
```
Comment (CharlesDuboisSAP) [inline docs-java_versioned_docs/version-v4/release-notes/release-notes-15-to-29.mdx:26]:
```suggestion
      - Update [Spring Framework](https://central.sonatype.com/artifact/org.springframework/spring-framework-bom/5.3.34) (`org.springframework:spring-framework-bom`) from `5.3.32` to `5.3.34`
      - Update [Lombok](https://central.sonatype.com/artifact/org.projectlombok/lombok/1.18.32) (`oorg.projectlombok: lombok `) from `1.18.30` to `1.18.32`
```

**Review by CharlesDuboisSAP — CHANGES_REQUESTED** [review]

File: docs-java_versioned_docs/version-v4/release-notes/release-notes-15-to-29.mdx (hunk 3 above)
Comment (newtork) [reply docs-java_versioned_docs/version-v4/release-notes/release-notes-15-to-29.mdx:26]: I would not mention Lombok. It's a dependency at compile time (i.e. `provided` scope in parent-pom).
Only exception would be generator modules. but IMHO this doesn't qualify.

**Review by CharlesDuboisSAP — APPROVED** [review]

**Pull request merged**